bundles/edu.kit.ipd.sdq.activextendannotations/src/edu/kit/ipd/sdq/activextendannotations/StaticDelegate.xtend

```diff
@@ -1,8 +1,382 @@
+			val validAnnotations = new ArrayList(annotations.length);
+			val usedTypes = new HashMap<Type, Set<AnnotationReference>>
+			for (annotation : annotations) {
+				val containedDuplicates = annotation.delegationTargets.map[type].duplicates
+				if (containedDuplicates.length > 0) {
+					containedDuplicates.forEach [
+						annotation.delegationTargetsExpression.
+							addWarning('''Type «simpleName» is listet multiple times.''')
+					]
+				}
+				val annotatedTypes = new HashSet(annotation.delegationTargets.map[type])
+				val duplicates = usedTypes.keySet.intersect(annotatedTypes)
+				if (duplicates.length == 0) {
+					validAnnotations += annotation
+				} else {
+					duplicates.forEach [ type |
+						(usedTypes.get(type) + #[annotation]).map[delegationTargetsExpression].forEach [
+							addError('''The type «type.simpleName» is also used in another @«StaticDelegate.simpleName» annotation.''')
+						]
+					]
+				}
+				annotatedTypes.forEach [ type |
+					usedTypes.computeIfPresent(type, [k, set|set += annotation; set]);
+					usedTypes.putIfAbsent(type, new HashSet(#[annotation]))
+				]
+			}
+			validAnnotations
+		}
+
+		def private checkForSelfReference(Iterable<? extends AnnotationReference> annotations,
+			MutableTypeDeclaration annotatedType) {
+			val validAnnotations = new ArrayList(annotations.length);
+			for (annotation : annotations) {
+				if (annotation.delegationTargets.map[type].contains(annotatedType)) {
+					annotation.getExpression('value').
+						addError('''A type cannot export its own methods. Remove the type «annotatedType.simpleName» from the list.''')
+				} else {
+					validAnnotations += annotation
+				}
+			}
+			validAnnotations
+		}
+
+		/**
+		 * Converts a {@link Visibility} instance to Xtend’s
+		 * {@link org.eclipse.xtend.lib.macro.declaration.Visibility}. Uses the
+		 * provided `defaultVisibility` if the instance is 
+		 * {@link Visibility.AS_DECLARED}.
+		 * 
+		 * @param visibility The visibility to convert.
+		 * @param defaultVisibility The visibility to use if `visibility` is
+		 * {@link Visibility.AS_DECLARED}
+		 */
+		def private static toXtendVisibility(Visibility visibility,
+			org.eclipse.xtend.lib.macro.declaration.Visibility defaultVisibility) {
+			switch (visibility) {
+				case PRIVATE:
+					org.eclipse.xtend.lib.macro.declaration.Visibility.PRIVATE
+				case PACKAGE: {
+					org.eclipse.xtend.lib.macro.declaration.Visibility.DEFAULT
+				}
+				case PUBLIC: {
+					org.eclipse.xtend.lib.macro.declaration.Visibility.PUBLIC
+				}
+				case AS_DECLARED: {
+					defaultVisibility
+				}
+			}
+		}
+
+		/**
+		 * Finds the duplicate entries in the provided `iterable`.
+		 * 
+		 * @param iterable The iterable to inspect
+		 * @return The set of elements present at least twice in the provided
+		 * `iterable`. Equality is checked based on {@link Object#equals}.
+		 */
+		def private static <T> duplicates(Iterable<T> iterable) {
```

Comment:
This project does not yet depend on the SDQ Commons and I thought it to be more likely to make the SDQ Commons depend on this project than the other way around (and from my perspective, circular dependencies should be avoided). That’s why I did not add this method to a more meaningful place.

Considering this, should the method still be moved to the SDQ Commons?